### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by nurbolsultanov**
Created a method called ``current_user`` in ApplicationController to make current user data available to all controllers ✔️
Created forms to perform the following functions: ✔️
- Creates a Post on behalf of the ``current_user``
- Create a comment on behalf of the ``current_user``

Added option to allow users to add likes to Posts ✔️

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,4 +11,21 @@ def show
...
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new, alert: 'Failed to create post.'
+    end
+  end
```
Comment (codepantha) [inline app/controllers/posts_controller.rb]: - [x] According to the requirements, a post should be created on behalf of the currently logged in user (`current_user`) of the app. You are currently setting the current user based on the `user_id` params. Please make this change 👍

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -6,19 +6,37 @@
...
+    <div class="field">
+      <%= form.label :title %>
+      <%= form.text_field :title %>
+    </div>
+    <div class="field">
+      <%= form.label :text %>
+      <%= form.text_area :text %>
+    </div>
+    <%= form.submit "Create Post" %>
+  <% end %>
```
Comment (codepantha) [inline app/views/posts/index.html.erb]: - [x] Please relocate this form for creating a post to the `views/posts/new.html.erb` file.
- [x] Also note that you would have to create a `new` method in your `PostsController` which will serve up the view to create a new post.

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -19,5 +23,13 @@
...
 <% end %>
   </ul>
+  <h3>Create a Comment</h3>
+<%= form_with(model: [@post, @post.comments.build], url: user_post_comments_path(@post.author, @post)) do |form| %>
+  <div class="field">
+    <%= form.label :text %>
+    <%= form.text_area :text %>
+  </div>
+  <%= form.submit "Create Comment" %>
+<% end %>
```
Comment (codepantha) [inline app/views/posts/show.html.erb:33]: - [x] Please relocate this form for creating a comment to the `views/comments/new.html.erb` file.
- [x] Also note that you would have to create a `new` method in your `CommentsController` which will serve up the view to create a new comment.
- [x] Kindly note that you have to create a route in your routes file for the `new` action
```ruby
GET   /posts/:post_id/comments/new
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @everinurmind 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added forms for creating posts and comments 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @everinurmind  👋 

### Things to appreciate
- [x] Worked on the missing things requested by the previous code reviewer. 👍 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**